js/js_test.go

```diff
@@ -571,3 +572,207 @@ func TestUint8Array(t *testing.T) {
 		t.Errorf("Non-empty byte array is not externalized as a Uint8Array")
 	}
 }
+
+// Internalize is used as a helper type to test $internalize. A struct is used
+// in order that all of the Go types can be verified (otherwise we are limited
+// to the methods on *Object). Where methods on *Object exist they too will be
+// tested
+type Internalize struct {
+	*js.Object
+
+	string string `js:"string"`
+	bool   bool   `js:"bool"`
+}
+
+func TestInternalizeString(t *testing.T) {
+	fieldName := "string"
+	zero := ""
+
+	s := &Internalize{Object: js.Global.Get("Object").New()}
```

Comment:
You raise a good point. I half considered this myself, but postponed giving it much thought until we've confirmed we're actually testing the right stuff... at which point extracting to table tests should be relatively easy. 

The little thought I did give it however amount to the following: I'm not convinced we _can_ actually use table tests. We need to address specific fields of the `Internalize` struct and do so in such a way that the `gopherjs` compiler sees us doing so and compiles to the corresponding Javascript. Put another way, the code we write in these tests needs to correspond _exactly_ to the code that will be written in the wild.

Therefore the one approach I have considered fairly seriously is code generating the tests. Because they do follow a formula that's for sure.

But as I said, I haven't given this much time/though so would very much welcome any suggestions.